lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -140,6 +142,25 @@ def scrub(transcript)
 
       private
 
+      def add_external_three_ds(params, payment_method, options)
+        return if options[:three_d_secure].blank?
+
+        three_d_secure_data = {}
+        three_d_secure_data[:program_protocol] = options[:three_d_secure][:version] if options[:three_d_secure][:version].present?
+        three_d_secure_data[:directory_server_transaction_id] = options[:three_d_secure][:ds_transaction_id] if options[:three_d_secure][:ds_transaction_id].present?
+        three_d_secure_data[:cardholder_name] = payment_method.name
+        three_d_secure_data[:card_number] = payment_method.number
+        three_d_secure_data[:exp_date] = format_exp_date(payment_method.month, payment_method.year)
+        three_d_secure_data[:cvv] = payment_method.verification_value if payment_method.verification_value?
+        three_d_secure_data[:xid] = options[:three_d_secure][:acs_transaction_id]
```

Comment:
❓ Questions:

1) does Payeeze supports 3DS version 1 and 2 (1 is deprecated but I'm seeing `xid` param there)
2) So are you going to send at the same time `directory_server_transaction_id` (3DS v1) and `xid` (3DS v2)?

Reply:
1. According to information in [link](https://developer.payeezy.com/payeezy-api/apis/post/transactions-7) payeezy supports both versions.
2.